### PR TITLE
test: address a number of unused result warnings

### PR DESCRIPTION
**Pull request by compnerd**
Address the various set of unused result warnings in the test suite.  NFC.

### CODE REVIEW

File: TestFoundation/TestNSBundle.swift [hunk 1]
```diff
@@ -115,12 +115,12 @@ class TestNSBundle : XCTestCase {
             
             // Put some resources in the bundle
             for n in _bundleResourceNames {
-                NSFileManager.defaultManager().createFile(atPath: bundlePath + "/" + n, contents: nil, attributes: nil)
+                _ = NSFileManager.defaultManager().createFile(atPath: bundlePath + "/" + n, contents: nil, attributes: nil)
```
Comment (parkera) [inline TestFoundation/TestNSBundle.swift]: I wonder if we should annotate this API as a discardable result instead. It doesn't have the error handling anyway...

File: TestFoundation/TestNSString.swift [hunk 2]
```diff
@@ -401,7 +401,7 @@ class TestNSString : XCTestCase {
         var buf : [UInt8] = []
         buf.reserveCapacity(reservedLength)
         var usedLen : CFIndex = 0
-        buf.withUnsafeMutableBufferPointer { p in
+        _ = buf.withUnsafeMutableBufferPointer { p in
```
Comment (parkera) [inline TestFoundation/TestNSString.swift]: I think the right place to ignore this result is actually the CFStringGetBytes (which is returning a value passed through the closure here)

File: TestFoundation/TestNSBundle.swift (hunk 1 above)
Comment (compnerd) [inline TestFoundation/TestNSBundle.swift]: Im happy to make such a change.  I imagine that a) that would need to go through swift evolution b) should be a follow up change.

File: TestFoundation/TestNSString.swift (hunk 2 above)
Comment (compnerd) [inline TestFoundation/TestNSString.swift]: Hmm, I think that I agree with you.  I will adjust the patch and upload it again.  Good thought!

File: TestFoundation/TestNSBundle.swift (hunk 1 above)
Comment (parkera) [inline TestFoundation/TestNSBundle.swift]: Let's file a follow up JIRA for it.

Comment (compnerd) [inline TestFoundation/TestNSBundle.swift]: https://bugs.swift.org/browse/SR-1638 ... can't seem to add you to it :-(.

**Comment by compnerd** [comment]
ping

**Comment by phausler** [comment]
Overall the changes look reasonable to me. I will take a look at this later today to see if we can get it merged manually

**Comment by compnerd** [comment]
Awesome!  Thanks!

**Pull request merged**